### PR TITLE
Fetch the common elements for the page

### PR DESCRIPTION
**Pull request by michalczaplinski**
Closes: #37

### CODE REVIEW

**Comment by frontibotito** [comment]
Deploy preview for _website_ ready!

Built with commit 973b4300abf42c71c7a9618635f816df709cadcc

https://frontity-org-5hosa0d65.now.sh

**Comment by michalczaplinski** [comment]
@SantosGuillamot @luisherranz 

This is my attempt at fetching the data for the custom elements. Now, I receive the 404 Error.

It seems to me that:
1. It should be more clearly documented how the "Custom Post Types" map to the URL that are used to fetch them. I still don't know if I should use `wp_template_parts` or `template-parts` and what it the relationship between them.
2. It should be more clear if the user (me 😅) needs do **some** or **all** of the following:
  a) Add a custom post type in `frontity.settings.js`
  b) Fetch the data in `beforeSSR`
  c) Create a handler for that post type ?
At the moment I don't know which ones are necessary 🙂

**Comment by SantosGuillamot** [comment]
Okay, maybe we need to update the docs. I'll try to explain it here to check what could be improved. When you add a Custom Post Type (CPT) to `wp-source`, the goal is to work with them the same way we work with posts or pages, fetching by the WordPress url instead of an specific API endpoint.  

There are two mandatory params: `type` and `endpoint`. That part is explained with an image in [the docs](https://docs.frontity.org/api-reference-1/wordpress-source#state-source-posttypes). Once you do that, you can fetch the content of a specific CPT using its url instead of going to the specific endpoint of the API (just like everything in Frontity). So, for example, if in your WordPress you would navigate to `/movies/terminator/`, in Frontity you can fetch directly the data doing `actions.source.fetch("/movies/terminator/")` (this fetch is done automatically by Frontity getting the current route).

That is how CPTs work with Frontity, and this way you don't need a handler, you just have to add some settings to `source`.

In this particular case, we want to get the data for the CPT "Template parts". In order to be able to get them by `url`, we have to add it to the `source` settings.
https___wp_frontity_org_wp-json_wp_v2_template-parts
Apart from that, we want to fetch all the elements added to `state.theme.templates`. Once we have added the CPT to the settings, we can fetch them using the url that would be used in WordPress: `actions.source.fetch("/wp_template_part/:slug/")`.

To sum up:
* We don't need a custom handler.
* It isn't working right now because the `endpoint` you added to the `settings` isn't correct.

Let me know if it's clear now and feel free to modify the docs with the aspects you consider that could be improved.

File: frontity.settings.js [hunk 1]
```diff
@@ -15,7 +15,13 @@ const settings = {
       name: "@frontity/wp-source",
       state: {
         source: {
-          api: "https://wp.frontity.org/wp-json"
+          api: "https://wp.frontity.org/wp-json",
+          postTypes: [
+            {
+              type: "wp_template_part",
+              endpoint: "wp_template_part"
```
Comment (SantosGuillamot) [inline frontity.settings.js]: The endpoint should be `template-parts`.

**Comment by luisherranz** [comment]
> Okay, maybe we need to update the docs.

@SantosGuillamot maybe it'd be nice if you could explain this to @juanmaguitar / @mburridge so they can add it to the docs roadmap :)

**Comment by michalczaplinski** [comment]
What is confusing to me still is this:

1. Why do we need to specify the `type` and `endpoint` in `frontity.settings.js` if I need to fetch it manually anyway like `actions.source.fetch('/wp_template_part/fixed-header')`?

2. The endpoint https://wp.frontity.org/wp-json/wp/v2/template-parts returns the data, but the one that is the "link" (http://wp.frontity.org/wp_template_part/fixed-header) returns a 404? What is the purpose of that "link"?

3. 
> When you add a Custom Post Type (CPT) to wp-source, the goal is to work with them the same way we work with posts or pages, fetching by the WordPress url instead of an specific API endpoint.

And according to our own docs: 
> Differentiating type and endpoint may be confusing as they are usually the same. You can confirm you are doing it correctly going to the CPT endpoint :

I don't understand this point. URLs are supposed to be unique anyways! Why do we need to have an "endpoint" and "type" which are usually the same can also be different...? 🤔

**Comment by SantosGuillamot** [comment]
I think it's important to understand what is a Custom Post Type and why it's useful in WordPress. [This detailed guide](https://www.wpbeginner.com/wp-tutorials/how-to-create-custom-post-types-in-wordpress/) could help.

Regarding your questions:
> 1. Why do we need to specify the type and endpoint in frontity.settings.js if I need to fetch it manually anyway like actions.source.fetch('/wp_template_part/fixed-header')?  What is the purpose of specifying the type and not just using the endpoint?

There are some aspects here:
* You specify the `type` and `endpoint` to make that CPT work by default in Frontity. This way you can work with urls instead of REST API endpoints. This means, for common CPT we can still use `state.source.get(state.router.link)` and work by default.
* It's true that this case is special because this CPT are elements and not something we want to access through a url, so I guess we could use just the endpoint, although it's not the way to work with CPT in Frontity.

>2. The endpoint https://wp.frontity.org/wp-json/wp/v2/template-parts returns the data correctly, but the one that is the "link" in the JSON that you posted (http://wp.frontity.org/wp_template_part/fixed-header) returns a 404? What is the purpose of that "link"?

The "link" is not working because we are redirecting the `wp.frontity.org` urls (excluding the REST API and the admin) to `blog.frontity.org`, where we are using Frontity. And in that Frontity project the CPT `wp_template_part` is not configured in `frontity.settings.js`. Anyway, we don't need it there as we aren't using it. In our case, if we define the CPT correctly, it won't return a 404.

> I don't understand this point. URLs are supposed to be unique anyways! Why do we need to have an "endpoint" and "type" which are usually the same but can also be different...?

When you defined a CPT in WordPress you have to fill some fields like `type`, `link` or `endpoint`. They have different purposes.
* `type`: How you want to name your new CPT. By default WordPress has different Post Types like `post`, `page` or `attachment`. But if, for example, you are running a movies website, you'd like to create a custom one named `movie`, so you can load different templates depending on if it's a post, page or a movie.
* `link`: You may want to access the all the movies with the pattern `/movies/:slug` so you can specify that as well.
* `endpoint`: You have to defined an endpoint where you can find all the movies, instead of `/wp-json/wp/v2/posts`.

So, inside Frontity:
* You define a `type` so you can do something like:
```
if (type === "movie"){
  <Movie />
} else {
  <Post />
}
```
* You define the `endpoint` to let Frontity knows where it should fetch the content of the movie.
* You use the `link` to fetch and get the content with `actions.source.fetch(link)`.

And all these fields can be different, that's the reason you have to define all of them.

**Comment by michalczaplinski** [comment]
@SantosGuillamot Thanks for the explanation, I think it makes it a bit more clear!
Can we review and merge this?

File: frontity.settings.js (hunk 1 above)
Comment (SantosGuillamot) [reply frontity.settings.js]: We still have to resolve this.

File: packages/frontity-org-theme/src/index.ts [hunk 2]
```diff
@@ -50,8 +60,33 @@ const frontityOrg: FrontityOrg = {
         borderRadius,
         dropdown
       ]
+    },
+    source: {
+      handlers: []
     }
```
Comment (SantosGuillamot) [inline packages/frontity-org-theme/src/index.ts]: As we are not using handlers, we can remove this field.

File: packages/frontity-org-theme/src/index.ts [hunk 3]
```diff
@@ -50,8 +60,33 @@ const frontityOrg: FrontityOrg = {
...
+    // 2. add product to state
+    const [template] = await libraries.source.populate({
+      response,
+      state,
+      force
+    });
+
+    console.log(template);
+  }
+});
```
Comment (SantosGuillamot) [inline packages/frontity-org-theme/src/index.ts]: With the approach we followed I think we don't need this handler anymore.

**Comment by michalczaplinski** [comment]
@SantosGuillamot ok, done.
I'm sorry I didnt realize that I havent pushed the updates yesterday when I asked for review.

**Comment by michalczaplinski** [comment]
@iamuchejude Could you review the PR?
